greenpeace/planet4-plugin-gutenberg-blocks
# PLANET-6771: Post blocks detection

Ref: https://jira.greenpeace.org/browse/PLANET-6771

Block assets are loaded if a post contains that specific block (ie: `ENFormStyle.min.css` is loaded if an `ENForm` block is used in the displayed post).
Block detection relies on `has_block()` native function, which does not check for reusable blocks content.
Cf. https://core.trac.wordpress.org/ticket/53140

This PR adds a new detection system based on `WP_Block_Parser`, that fixes this issue.

## Test

Either clone GPLux instance which has this issue on the frontpage, or 
- Create a reusable block with an enform block in it
- Use that reusable block in a new page

When displaying the page, check that `ENFormStyle.min.css` is loaded.

Also check the cache content, for example:
```
> docker-compose exec php-fpm wp cache get post-306 blocklist
```
should show a list of blocks used in the post 306.
